### ISSUE TITLE
wix-ui-core: LabelWithOptions issues and enhancements

### ISSUE THREAD

**Issue by shlomitc** [wix/wix-ui#494]
1. Keyboard navigation: should be relatively easy to make it work as it is using the Dropdown component that handles it. Just need to propagate the events correctly and add `tabIndex` for the Label component.

2. Add checkbox icons: need to pass in the `Checkbox` props to the component. should decide whether passing all props, or just an abstraction of props (only icons for example).

3. Since it uses `Dropdown` internally - Dropdown is already managing the multi-select state which is duplicated also in the `LabelWithOptions`. Should refactor `Dropdown` to propagate the entire selected options list instead of just a single selected option.
